src/js/wall/Reset.jsx

```diff
@@ -14,13 +14,22 @@ export class Reset extends React.Component {
         };
     }
 
+    componentDidMount() {
+        this.setState({
+            resetCode: this.props.resetCode,
+        });
+    }
+
     handleChange = e => {
         this.setState({ password: e.target.value });
     };
 
     handleSubmit = e => {
         e.preventDefault();
-        this.props.onReset(this.state.password, this.props.resetCode);
+
+        if (!this.state.password.length < 8) {
+            this.props.onReset(this.state.password, this.state.resetCode);
+        }
```

Comment:
Careful! This conditional always resolves to true as written
```suggestion
        if (!(this.state.password.length < 8)) {
            this.props.onReset(this.state.password, this.state.resetCode);
        }
```

Also: Will this render an error to the user if the request is blocked? I don't see it in the code, and I would hate to block the user but not say why